ui-manchette-with-spacetimechart/src/hooks/useManchetteWithSpaceTimeChart.ts

```diff
@@ -0,0 +1,218 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import type {
+  ProjectPathTrainResult,
+  StyledOperationalPointType,
+  OperationalPointType,
+} from '@osrd-project/ui-manchette/src/types';
+import type { OperationalPoint, SpaceScale } from '@osrd-project/ui-spacetimechart/dist/lib/types';
+
+import usePaths from './usePaths';
+import { MAX_ZOOM_Y, MIN_ZOOM_Y, ZOOM_Y_DELTA, INITIAL_SPACE_TIME_CHART_HEIGHT } from '../consts';
+import {
+  calcOperationalPointsToDisplay,
+  computeTimeWindow,
+  getOperationalPointsWithPosition,
+  getScales,
+  operationalPointsHeight,
+} from '../helpers';
+import { getDiff } from '../utils/vector';
+import { useIsOverflow } from './useIsOverFlow';
+
+type State = {
+  xZoom: number;
+  yZoom: number;
+  xOffset: number;
+  yOffset: number;
+  panning: { initialOffset: { x: number; y: number } } | null;
+  scrollPosition: number;
+  isProportional: boolean;
+  operationalPointsChart: OperationalPoint[];
+  operationalPointsToDisplay: StyledOperationalPointType[];
+  panY: boolean;
+  scales: SpaceScale[];
+};
+
+const useManchettesWithSpaceTimeChart = (
+  operationalPoints: OperationalPointType[],
+  projectPathTrainResult: ProjectPathTrainResult[],
+  manchette: React.RefObject<HTMLDivElement>,
+  selectedProjection?: number,
+  height = 561
+) => {
+  const [isShiftPressed, setIsShiftPressed] = useState(false);
+  const [state, setState] = useState<State>({
+    xZoom: 1,
+    yZoom: 1,
+    xOffset: 0,
+    yOffset: 0,
+    panning: null,
+    scrollPosition: 0,
+    isProportional: true,
+    operationalPointsChart: [],
+    operationalPointsToDisplay: [],
+    panY: false,
+    scales: [],
+  });
+
+  const { xZoom, yZoom, xOffset, yOffset, panY, panning, scrollPosition, isProportional } = state;
+
+  const paths = usePaths(projectPathTrainResult, selectedProjection);
+
+  const checkOverflow = useCallback((isOverflowFromCallback: boolean) => {
+    setState((prev) => ({ ...prev, panY: isOverflowFromCallback }));
+  }, []);
+  useIsOverflow(manchette, checkOverflow);
```

Comment:
I don't really get what this piece of code is used for. If I comment it, nothing changes. Do we need it ?

Reply:
It checked if the manchette got overflow, and if not it prevents the spaceTimeChart to being able to pan vertically